### ISSUE TITLE
Error Restoring Order Product

### ISSUE THREAD

**Issue by jsdanie1** [Murloc124/DataRestorationTool#10]
I know these restoration tools aren't very good with sub related data. But curious how to get past the error below. Trying to restore Order Products on the Order. This tool worked well restoring the Orders just fine, but can't seem to get the Order Products to restore. I thought it was because the tool imported all the status incorrectly, and i manually changed them to the correct status but still getting this error regardless. 

---
```
System.Exception:
A validation error occurred. The value 3 of 'salesorderstatecode' on record of type 'salesorderdetail' is outside the valid range. Accepted Values: 
Data Restoration Tool
- DataRestorationTool.MyPluginControl.HandleEMPReponse(ExecuteMultipleResponseItemCollection responses, Boolean contOnError, List`1 errorList)
- DataRestorationTool.MyPluginControl.<>c__DisplayClass26_0.<RestoreData>b__0(BackgroundWorker w, DoWorkEventArgs e)
```

---
- XrmToolBox Version: 1.2023.12.68
- DataRestorationTool Version: 1.2023.10.25
- DB Version: 9.2.24023.204
- Deployment: Online

**Comment by Murloc124** [comment]
Hi,
Thanks for the feedback!

I'm not entirely sure how this is currently handled as it blindly copies the Audit History's Old Value component to a new record.
I will perform some testing on this topic on my end as well to be able to further assist you on this.

Could you check in your environment if both the Statecode and Statuscode fields are enabled for auditing?

**Comment by jsdanie1** [comment]
Yep these fields are enabled for auditing by default for Dynamics 365.
